plugins/automelt.cpp

```diff
@@ -1,294 +1,705 @@
-#include "uicommon.h"
 
-#include "modules/Gui.h"
+#include "Debug.h"
+#include "LuaTools.h"
+#include "PluginManager.h"
+#include "TileTypes.h"
+
+#include "modules/Buildings.h"
+#include "modules/Maps.h"
+#include "modules/Items.h"
+#include "modules/World.h"
+#include "modules/Designations.h"
+#include "modules/Persistence.h"
+#include "modules/Units.h"
+
+// #include "uicommon.h"
 
 #include "df/world.h"
+#include "df/building.h"
 #include "df/world_raws.h"
 #include "df/building_def.h"
 #include "df/viewscreen_dwarfmodest.h"
 #include "df/building_stockpilest.h"
-#include "modules/Buildings.h"
-#include "modules/Items.h"
 #include "df/plotinfost.h"
-#include "modules/Maps.h"
-#include "modules/World.h"
 #include "df/item_quality.h"
 
+#include <map>
+#include <unordered_map>
+
 using df::building_stockpilest;
+using std::map;
+using std::multimap;
+using std::pair;
+using std::string;
+using std::unordered_map;
+using std::vector;
+
+using namespace DFHack;
+using namespace df::enums;
 
 DFHACK_PLUGIN("automelt");
-#define PLUGIN_VERSION 0.3
 REQUIRE_GLOBAL(gps);
 REQUIRE_GLOBAL(world);
 REQUIRE_GLOBAL(cursor);
 REQUIRE_GLOBAL(plotinfo);
 
-static const string PERSISTENCE_KEY = "automelt/stockpiles";
+namespace DFHack
+{
+    DBG_DECLARE(automelt, status, DebugCategory::LINFO);
+    DBG_DECLARE(automelt, cycle, DebugCategory::LINFO);
+    DBG_DECLARE(automelt, perf, DebugCategory::LTRACE);
```

Comment:
this sets the *default* log level to TRACE, which is almost certainly not what you want. This should probably be set to LINFO here. If you want trace output, you'd run `debugfilter set trace automelt perf`

Reply:
Ah, good to know. I need to add a few more debug calls to get timing info and to check perf. impact.